internal/clients/config/elasticsearch.go

```diff
@@ -132,6 +140,13 @@ func newElasticsearchConfigFromFramework(ctx context.Context, cfg ProviderConfig
 		config.Addresses = endpoints
 	}
 
+	if esConfig.BearerToken.ValueString() != "" || esConfig.BearerToken.ValueString() != "null" {
```

Comment:
```suggestion
	if esConfig.BearerToken.ValueString() != "" {
```

If the value is `null` in the TF module, ValueString will return `""`